sdks/python/apache_beam/runners/portability/stager.py

```diff
@@ -365,6 +368,16 @@ def create_job_resources(options,  # type: PipelineOptions
             Stager._create_file_stage_to_artifact(
                 pickled_session_file, names.PICKLED_MAIN_SESSION_FILE))
 
+    # stage the submission environment dependencies
+    local_dependency_file_path = os.path.join(
+        temp_dir, SUBMISSION_ENV_DEPENDENCIES_FILENAME)
+    dependencies = freeze.freeze()
```

Comment:
Let's make this best effort: if for whatever reason this fails, don't fail the job submission.

You could also consider moving this portion of code into a helper since this method keeps growing (no strong opinion).